sources/models/src/de.rs

```diff
@@ -0,0 +1,73 @@
+use crate::RegistryMirror;
+use serde::de::value::SeqAccessDeserializer;
+use serde::de::{MapAccess, SeqAccess, Visitor};
+use serde::{Deserialize, Deserializer};
+use std::fmt::Formatter;
+
+// We want to allow two possible representations of registry mirrors. Either as a `Vec` of `RegistryMirror`
+// or a `HashMap` of registry to endpoints.
```

Comment:
```suggestion
// Our standard representation of registry mirrors is a `Vec` of `RegistryMirror`; for backward compatibility, we also allow a `HashMap` of registry to endpoints.
```
(I wasn't sure where to break the line, but it's probably useful to make clear that we don't encourage the old format.)